velox/type/TimestampConversion.cpp

```diff
@@ -812,6 +827,121 @@ Status parserError(const char* str, size_t len) {
       std::string(str, len));
 }
 
+inline bool startsWith(std::string_view str, const char* prefix) {
+  return str.rfind(prefix, 0) == 0;
+}
+
+// Spark support an id with one of the prefixes UTC+, UTC-, GMT+, GMT-, UT+ or
+// UT-, and a suffix in the formats: h[h], hh[:]mm, hh:mm:ss, hhmmss. This func
+// will validate the format of the suffix.
+// Additionally, IANA does not support seconds in the offset, so this func will
+// add the seconds offset to the input Timestamp.
+bool parseSparkTzWithPrefix(
+    std::string& timeZoneName,
```

Comment:
Can we use 'std::string_view' as parameter type to avoid creating intermediate strings?

Reply:
This func will return bool to indicate whether an exception has occurred. Using `std::string&` for the parameter is because we also need to modify the content of `timeZoneName`.